File: DesignPatternsExercises/java/VisitorDesignPattern/Exercise/Order.java
import java.util.ArrayList;

public class Order implements Visitable {
    private String name;
    private ArrayList<String> items = new ArrayList<>();

    public Order(String name) {
        this.name = name;
    }

    @Override
    public void accept(Visitor visitor) {
        visitor.__________(7)__________;
    }

    public void addItem(String item) {
        items.add(item);
    }

    public String getName() {
        return name;
    }

    public ArrayList<String> __________(8)__________() {
        return items;
    }
}

File: DesignPatternsExercises/ruby/visitor_design_pattern/exercise/order.rb
require_relative 'visitable'

class Order < Visitable
  attr_reader __________(9)__________, :items

  def initialize(name)
    @name = name
    __________(10)__________ = []
  end

  def __________(11)__________(visitor)
    visitor.__________(12)__________
  end

  def add_item(item)
    @items << item
  end
end
